AnastasiyaSantalova/hw3.js

```diff
@@ -0,0 +1,308 @@
+"use strict"
+
+// FUNCTION 1: returns phrase with determined separator (via reduce)
+
+function splitAndMerge (str, sp) {
+
+    //separate phrase into words
+    var arrOfWords = str.split(" ");
+
+    var arrOfLetters = arrOfWords.map(function(word) {
+
+        //separate words into letters
+        var letters = word.split("");
+
+        //add new separator between letters
+        return letters.reduce(function(acc, letter) {
+            return acc + sp + letter;
+        });
+    });
+
+    //add new separator between words
+    return arrOfLetters.reduce(function(acc, word) {
+        return acc + sp + word;
```

Comment:
Actually, I don't understand, what do you mean

Reply:
Well, in the final string words containing your custom separator should be separated by whitespaces itself like `H.e.l.l.o w.o.r.l.d`, but in your case it will be `H.e.l.l.o.w.o.r.l.d` since you use the same separator to divide words itself

Reply:
Exactly) Fixed!